### PR TITLE
npm Migration 2: Reject installation for newer Wasp versions

### PR DESCRIPTION
**Pull request by cprecioso**
## Summary

Prevent installer from installing Wasp 0.21 and later, which should be installed via npm instead.

## Changes

- Add version checking with `version_gte()` function
- Reject installation attempts for Wasp 0.21+
- Display error message with npm installation instructions
- Display warning when installing older versions

## Part of stacked PRs

This is part 2 of 5:
1. #15
2. **This PR**
3. #17
4. #18
5. #19

### CODE REVIEW

File: installer.sh [hunk 1]
```diff
@@ -49,6 +50,18 @@ main() {
...
+    # Require version argument if specified in VERSION_ARG
+    if [ -n "$VERSION_ARG" ]; then
+        # Check version restrictions - reject when requested version >= migration version
+        if version_gte "$VERSION_ARG" "$NPM_MIGRATION_VERSION"; then
+            die "Wasp version $NPM_MIGRATION_VERSION and later must be installed via npm.\n\nPlease run: npm install -g @wasp.sh/wasp-cli@$VERSION_ARG\n\nTo migrate from installer to npm, run:\n  curl -sSL https://get.wasp.sh/installer.sh | sh -s -- migrate-to-npm"
+        fi
+
+        # Warn about installing old version
+        info "${RED}WARNING${RESET}: You are installing an older version of Wasp ($VERSION_ARG)."
+        info "Starting with Wasp $NPM_MIGRATION_VERSION, the installer is deprecated and npm is the preferred installation method:\n  npm install -g @wasp.sh/wasp-cli\n"
```
Comment (FranjoMindek) [inline installer.sh]: This one would cause a user to error if they just copied the installed method, because we need to migrate first.

File: installer.sh [hunk 2]
```diff
@@ -49,6 +50,18 @@ main() {
         exit 0
     fi
 
+    # Require version argument if specified in VERSION_ARG
```
Comment (FranjoMindek) [inline installer.sh:59]: Okay I was about to say we don't require users to use the version arg, but I see that is the next PR, maybe order of PRs 2 and 3 is flipped :D.

File: installer.sh [hunk 3]
```diff
@@ -49,6 +50,18 @@ main() {
...
+    if [ -n "$VERSION_ARG" ]; then
+        # Check version restrictions - reject when requested version >= migration version
+        if version_gte "$VERSION_ARG" "$NPM_MIGRATION_VERSION"; then
+            die "Wasp version $NPM_MIGRATION_VERSION and later must be installed via npm.\n\nPlease run: npm install -g @wasp.sh/wasp-cli@$VERSION_ARG\n\nTo migrate from installer to npm, run:\n  curl -sSL https://get.wasp.sh/installer.sh | sh -s -- migrate-to-npm"
```
Comment (FranjoMindek) [inline installer.sh]: Yes this is a bit tricky because we have to migrate first and then run the install, but we specify them in the opposite order.

Maybe something along the lines?

```suggestion
            die "Wasp version $NPM_MIGRATION_VERSION and later must be installed via npm.\n\nIf you've already installed Wasp from installer, please migrate to npm first via:\n curl -sSL https://get.wasp.sh/installer.sh | sh -s -- migrate-to-npm\n\nTo install Wasp over npm, please run:\n npm install -g @wasp.sh/wasp-cli@$VERSION_ARG
```

We can always link to docs for more details / instead of explaning it here.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -105,3 +105,40 @@ jobs:
...
+
+    steps:
+      - uses: actions/checkout@v5
+
+      - name: Install shell
+        if: runner.os == 'Linux'
+        run: sudo apt-get install -y ${{ matrix.shell }}
+
+      - name: Try to install version ${{ matrix.version }} (should fail)
+        run: |
```
Comment (FranjoMindek) [inline .github/workflows/test.yml:143]: Should we also test that old versions work with specified version e.g. `0.18`, `0.20`?

File: installer.sh [hunk 5]
```diff
@@ -11,6 +11,7 @@ HOME_LOCAL_BIN="$HOME/.local/bin"
 HOME_LOCAL_SHARE="$HOME/.local/share"
 WASP_LANG_DIR="$HOME_LOCAL_SHARE/wasp-lang"
 NPM_MARKER_FILE="$WASP_LANG_DIR/.uses-npm"
+NPM_MIGRATION_VERSION="0.21" # First version we'll refuse to install
```
Comment (FranjoMindek) [inline installer.sh]:
```suggestion
NPM_MIGRATION_VERSION="0.21" # First version we'll refuse to install through installer
```

or

```suggestion
NPM_MIGRATION_VERSION="0.21" # First version that requires installing through npm
```

**Review by FranjoMindek — APPROVED** [review]
Nothing blocking again, just minor suggestions.
Only thing is that PR 3 should've went before PR 2.

It's hard to follow the logic to be when it's not yet there.

I think there is no need for too much discussion here, so I will approve immediately.

File: installer.sh (hunk 2 above)
Comment (cprecioso) [reply installer.sh:59]: 🤦‍♂️ sorry, indeed maybe i could have split the other way around

File: .github/workflows/test.yml (hunk 4 above)
Comment (cprecioso) [reply .github/workflows/test.yml:143]: this is already being tested at https://github.com/wasp-lang/get-wasp-sh/blob/98c89a27e95437f2768ff636b4ddb1531375276c/.github/workflows/test.yml#L32, with v0.18. Should I add 0.20 to the matrix?

Comment (FranjoMindek) [reply .github/workflows/test.yml:143]: Aha okay, it should be at least one test, so this is fine.
I didn't know we had it since it's not part of the diffs.

You could change the `0.18` to `0.20` if you want to test the edge case specifically.
Not a requirement though.
